<launch>

	<!-- to be used with robot config: torso = back -->

	<!-- send urdf to param server -->
	<param name="bottom_plane" textfile="$(find cob_surface_classification)/models/objects/bottom_plane.urdf" />
	<!-- push urdf to factory and spawn robot in gazebo -->
	<node name="spawn_bottom_plane" pkg="gazebo_ros" type="spawn_model" args=" -param bottom_plane -urdf -x -10.5 -y 0 -z 0  -R 0 -P 0 -Y 0 -model bottom_plane" respawn="false" output="screen" />

	<param name="floor" textfile="$(find cob_surface_classification)/models/objects/floor.urdf" />
	<node name="spawn_floor" pkg="gazebo_ros" type="spawn_model" args=" -param floor -urdf -x -1.3 -y 0 -z -0.1 -R 0 -P 0 -Y 0 -model floor" respawn="false" output="screen" />

	<param name="wall" textfile="$(find cob_surface_classification)/models/objects/wall.urdf" />
	<node name="spawn_wall" pkg="gazebo_ros" type="spawn_model" args=" -param wall -urdf -x -3.1 -y 0 -z 0.15 -R 0 -P 0 -Y 0 -model wall" respawn="false" output="screen" />


	<param name="cylinder" textfile="$(find cob_surface_classification)/models/objects/cylinder.urdf" />
	<node name="spawn_cylinder" pkg="gazebo_ros" type="spawn_model" args=" -param cylinder -urdf -x -1.2 -y 0.2 -z 0.65  -R 0 -P 0 -Y 0 -model cylinder" respawn="false" output="screen" />

	<param name="cone" textfile="$(find cob_surface_classification)/models/objects/cone.urdf" />
	<node name="spawn_cone" pkg="gazebo_ros" type="spawn_model" args=" -param cone -urdf -x -1.3 -y -0.1 -z 0.65  -R 0 -P 0 -Y 0 -model cone" respawn="false" output="screen" />

	<param name="cylinder_hollow" textfile="$(find cob_surface_classification)/models/objects/cylinder_hollow.urdf" />
	<node name="spawn_cylinder_hollow" pkg="gazebo_ros" type="spawn_model" args=" -param cylinder_hollow -urdf  -x -1.1 -y -0.4 -z 0.65  -R 0 -P 0 -Y 0 -model cylinder_hollow" respawn="false" output="screen" />

</launch>
